tests/pages/multipleOperatorsServiceList.test.tsx

```diff
@@ -0,0 +1,252 @@
+                serviceCode: 'NW_05_BLAC_X1_1',
+            },
+            {
+                lineName: 'Infinity Line',
+                startDate: '07/02/2020',
+                description: 'This is some kind of bus service',
+                serviceCode: 'WY_13_IWBT_07_1',
+            },
+        ];
+
+        beforeEach(() => {
+            (getServicesByNocCode as jest.Mock).mockImplementation(() => mockServices);
+        });
+
+        it('should render correctly', () => {
+            const tree = shallow(
+                <MultipleOperatorsServiceList
+                    serviceList={mockServiceList}
+                    buttonText="Select All"
+                    errors={[]}
+                    csrfToken=""
+                    pageProps={[]}
+                    operatorName="Test Operator"
+                    nocCode="TO"
+                />,
+            );
+            expect(tree).toMatchSnapshot();
+        });
+
+        it('should render correctly with errors', () => {
+            const tree = shallow(
+                <MultipleOperatorsServiceList
+                    serviceList={mockServiceList}
+                    buttonText="Select All"
+                    errors={mockError}
+                    csrfToken=""
+                    pageProps={[]}
+                    operatorName="Test Operator"
+                    nocCode="TO"
+                />,
+            );
+            expect(tree).toMatchSnapshot();
+        });
+
+        describe('getServerSideProps', () => {
+            it('should return expected props to the page when the page is first visited by the user', async () => {
+                const ctx = getMockContext({
+                    session: {
+                        [MULTIPLE_OPERATOR_ATTRIBUTE]: {
+                            selectedOperators: [
+                                { operatorPublicName: 'Test1', nocCode: 'N1' },
+                                { operatorPublicName: 'Test2', nocCode: 'N2' },
+                                { operatorPublicName: 'Test3', nocCode: 'N3' },
+                            ],
+                        },
+                    },
+                });
+                const result = await getServerSideProps(ctx);
+                const expectedCheckedServiceList: ServicesInfo[] = mockServices.map(mockService => ({
+                    ...mockService,
+                    checked: false,
+                }));
+                expect(result.props.errors.length).toBe(0);
+                expect(result.props.serviceList).toEqual(expectedCheckedServiceList);
+                expect(result.props.buttonText).toEqual('Select All Services');
+                expect(result.props.operatorName).toBe('Test1' || 'Test2' || 'Test3');
+                expect(result.props.nocCode).toBe('N1' || 'N2' || 'N3');
+            });
+            it('should return expected props to the page when the page is visited by the user for a second time', async () => {
```

Comment:
Can we add newlines between these tests?

Reply:
Done.